File: php/src/Framework/Database/SqliteTransactionManager.php
<?php
declare(strict_types=1);

namespace Demo\App\Framework\Database;

class SqliteTransactionManager implements TransactionManager
{
    private DatabaseConnection $dbConnection;

    public function __construct(DatabaseConnection $dbConnection)
    {
        $this->dbConnection = $dbConnection;
    }

    public function beginTransaction(): void
    {
        $this->dbConnection->execute('BEGIN TRANSACTION;');
    }

    public function commit(): void
    {
        $this->dbConnection->execute('COMMIT;');
    }

    public function rollback(): void
    {
        $this->dbConnection->execute('ROLLBACK;');
    }
}

File: kotlin/src/main/kotlin/framework/database/SqliteTransactionManager.kt
package framework.database

class SqliteTransactionManager(private val dbConnection: DatabaseConnection) : TransactionManager {

    override fun beginTransaction() {
        dbConnection.execute("BEGIN TRANSACTION;")
    }

    override fun commit() {
        dbConnection.execute("COMMIT;")
    }

    override fun rollback() {
        dbConnection.execute("ROLLBACK;")
    }
}